# Plugin
local_planner: field_local_planner::RmpPlugin

# Base settings
requires_sensing: true # it should be true for all the local planners
control_rate: 10 # in Hz, rate at which the control command is computed
base_inverted: false # to flip the orientation of the robot 180 deg

# Robot size
robot_length: 1.0
robot_width: 0.5
robot_height: 0.5

# Frames
base_frame: base
fixed_frame: odom

# Thresholds
distance_to_goal_thr: 0.1 # meters
orientation_to_goal_thr: 0.1  # radians

# Velocity limits
max_linear_velocity_x: 1.0    # m/s
max_linear_velocity_y: 1.0    # m/s
max_angular_velocity_z: 1.0   # rad/s

# Other parameters
rmp:
  # Tune the sphere radius used for the control points
  sphere_radius_factor: 1.2

  # RMP computes an optimal acceleration (force) that
  # Must be integrated to get a velocity
  # This basically scales the acceleration
  integration_time: 0.1
  
  # Control points define the points where the accelerations are exerted on the body
  control_points:
    -
      id: center
      point_factor: [0.0, 0.0]
      color: [0.00, 0.00, 0.00]
      radius: 0.35
      affected_by:
        - goal_position
        - goal_orientation
    -
      id: front_left
      point_factor: [0.6, 0.3]
      color: [0.00, 0.00, 0.00]
      radius: 0.3
      affected_by: 
        - sdf_obstacle
    -
      id: front_right
      point_factor: [0.6, -0.3]
      color: [0.00, 0.00, 0.00]
      radius: 0.3
      affected_by: 
        - sdf_obstacle
    -
      id: back_left
      point_factor: [-0.6, 0.3]
      color: [0.00, 0.00, 0.00]
      radius: 0.3
      affected_by: 
        - sdf_obstacle
    -
      id: back_right
      point_factor: [-0.6, -0.3]
      color: [0.00, 0.00, 0.00]
      radius: 0.3
      affected_by: 
        - sdf_obstacle
      
  # This defines the parameters for the RMPs
  rmps:
    geodesic_goal:
      weight: 1.0
      gain: 20.0
      color: [1.0, 1.0, 0.0]
      metric:
        type: constant
        offset: 0.0
        steepness: 1.0
    geodesic_heading:
      weight: 1.0
      gain: 20.0
      color: [1.0, 1.0, 0.0]
      metric:
        type: constant
        offset: 0.0
        steepness: 1.0
    goal_position:
      weight: 0.0
      gain: 20.0
      color: [1.0, 0.5, 0.0]
      metric:
        type: constant
        offset: 0.0
        steepness: 1.0
    goal_orientation:
      weight: 1.0
      gain: 20.0
      color: [1.0, 0.5, 0.0]
      metric:
        type: constant
        offset: 0.0
        steepness: 1.0
    velocity_heading:
      weight: 0.0
      gain: 10.0
      color: [0.0, 1.0, 0.0]
      metric:
        type: constant
        offset: 0.0
        steepness: 1.0
    damping:
      weight: 1.0
      gain: 0.5
      color: [0.0, 0.0, 1.0]
      metric:
        type: constant
        offset: 0.0
        steepness: 1.0
    sdf_obstacle:
      weight: 1.0
      gain: 1.0
      color: [1.0, 0.0, 0.0]
      metric:
        type: constant
        offset: 0.0
        steepness: 1.0
    regularization:
      weight: 1.0
      gain: 1e-6
      color: [0.0, 0.0, 0.0]
      metric:
        type: constant
        offset: 0.0
        steepness: 1.0
